datasets/common_voice/common_voice.py

```diff
@@ -0,0 +1,797 @@
+        self.num_of_voice = kwargs.pop("num_of_voice", None)
+        description = f"Common Voice speech to text dataset in {self.language} version {self.sub_version} of {self.date_of_snapshot}. The dataset comprises {self.validated_hr_total} of validated transcribed speech data from {self.num_of_voice} speakers. The dataset has a size of {self.size}"
+        super(CommonVoiceConfig, self).__init__(name=name, version=datasets.Version("6.1.0", ""), description=description, **kwargs)
+
+
+# TODO: Name of the dataset usually match the script name with CamelCase instead of snake_case
+class CommonVoice(datasets.GeneratorBasedBuilder):
+    """TODO: Short description of my dataset."""
+
+    @property
+    def manual_download_instructions(self):
+        return """\
+    You need to go to https://commonvoice.mozilla.org/en/datasets,
+    and manually download the dataset as a .tar file. Once it is completed,
+    a folder will be made containing the files, validated.tsv, train.tsv, test.tsv, reported.tsv, other.tsv, invalidated.tsv, dev.tsv
+    and the folder clips containing audiofiles sampled at 48khz. Each clip is around 3-4 seconds in duration with a size of around 20-50 khz"""
+
+    BUILDER_CONFIGS = [
+        CommonVoiceConfig(
+            name=lang, 
+            sub_version=_LANGUAGES[lang]["Version"], 
+            date=_LANGUAGES[lang]["Date"], 
+            size=_LANGUAGES[lang]["Size"], 
+            val_hrs=_LANGUAGES[lang]["Validated_Hr_Total"], 
+            total_hrs=_LANGUAGES[lang]["Overall_Hr_Total"], 
+            num_of_voice=_LANGUAGES[lang]["Number_Of_Voice"], 
+        )
+        for lang in _LANGUAGES.keys()
+    ]
+
+#    DEFAULT_CONFIG_NAME = "first_domain"  # It's not mandatory to have a default configuration. Just use one if it make sense.
+
+    def _info(self):
+        # TODO: This method specifies the datasets.DatasetInfo object which contains informations and typings for the dataset
+#        if self.config.name == "first_domain":  # This is the name of the configuration selected in BUILDER_CONFIGS above 
+#            features = datasets.Features(
+#                {
+#                    "client_id": datasets.Value("string"),
+#                    "path": datasets.Value("string"),
+#                    "sentence": datasets.Value("string"),
+#                    "up_votes": datasets.Value("int64"),
+#                    "down_votes": datasets.Value("int64"),
+#                    "age": datasets.Value("string"),
+#                    "gender": datasets.Value("string"),
+#                    "locale": datasets.Value("string"),
+#                    "segmet": datasets.Value("string"),
+#
+                    # These are the features of your dataset like images, labels ...
+#                }
+#            )
+#
+#        else:  # This is an example to show how to have different features for "first_domain" and "second_domain"
+        features = datasets.Features(
```

Comment:
I think we should always have the same feature names -> are there languages for which these features won't work?